### PR TITLE
Dataset/View stats bugs

### PR DESCRIPTION
**Pull request by tylerganter**
1. The app was not displaying updated stats on the sidebar when a view was added.
2. classifications and detections now show the number of classifications/detections, not the number of samples with classifications/detections

 - [x] TODO this doesn't appear to update when the view is changed, so I need to fix that

### CODE REVIEW

**Comment by tylerganter** [comment]
@benjaminpkane when views are changed on the app, these stats are not being updated. when things are changed in python it updates correctly. do you have a fix for this?

**Comment by benjaminpkane** [comment]
Looking into it. One moment.

**Comment by benjaminpkane** [comment]
Do things look better now? (after https://github.com/voxel51/fiftyone/pull/404/commits/0cd5f81bee756d2d659bb83bc5e4f0e44b2b1921)

**Comment by tylerganter** [comment]
> Do things look better now? (after [0cd5f81](https://github.com/voxel51/fiftyone/commit/0cd5f81bee756d2d659bb83bc5e4f0e44b2b1921))

Aha, that was the bug i just introduced. so now the app is working, but if I create/modify/delete a stage in the app it doesn't update the view stats

**Comment by benjaminpkane** [comment]
OK. Give me moment, I'll get back to you on this.

**Comment by benjaminpkane** [comment]
OK. Fixed now.

**Comment by benjaminpkane** [comment]
I'm aware that this change isn't optimal, but I'll defer to you on changes.

I can make the ViewBar send a response without `derivables`, but that doesn't change the fact that the session is waiting on a query for data it doesn't need (but the app does need it).

File: fiftyone/core/state.py [hunk 1]
```diff
@@ -113,10 +116,7 @@ class StateDescriptionWithDerivables(StateDescription):
     def __init__(self, derivables=None, *args, **kwargs):
```
Comment (tylerganter) [inline fiftyone/core/state.py:116]: can we remove `derivables` from the `__init__()` signature then?

Comment (benjaminpkane) [reply fiftyone/core/state.py:116]: Yes, missed that part. Sounds good.

**Comment by tylerganter** [comment]
Okay I removed the unused arg. I just played around with the app and it appears to be working well!

File: fiftyone/core/state.py [hunk 2]
```diff
@@ -201,18 +205,59 @@ def get_dataset_stats(dataset):
...
+
+    _sample_doc_cls = type(view.dataset_name, (foo.DatasetSampleDocument,), {})
     num_default_fields = len(_sample_doc_cls.get_field_schema())
 
-    field_names = [field_name for field_name in dataset.get_field_schema()]
+    field_schema = view.get_field_schema()
+    custom_fields_schema = OrderedDict(
+        {
+            k: field_schema[k]
+            for k in list(field_schema.keys())[num_default_fields:]
```
Comment (benjaminpkane) [inline fiftyone/core/state.py:220]: @tylerganter the way custom fields are retrieved is not deterministic. Requesting that these keys are extracted in a way that all of the current current tests pass.

On the test that is failing, I'm seeing custom fields vary.

Sometimes `metadata` and `filepath` end up as keys here in the python3.5 server tests.

Comment (tylerganter) [reply fiftyone/core/state.py:220]: thanks for diagnosing this @benjaminpkane . I'll work on a fix

Comment (tylerganter) [reply fiftyone/core/state.py]: interesting...so `field_schema` is an `OrderedDict` and in python 3.5 `list(field_schema.keys())` is not deterministic. I made a solution that does not rely on order

**Pull request merged**